### PR TITLE
Feature/docker containerize

### PR DESCRIPTION
**Pull request by hendraustin**
Introduce Dockerfile into project to allow for building and running in Docker environment. Also, added Makefile for quality of life commands.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+**/Dockerfile*
+**/node_modules
+**/npm-debug.log
+**/obj
+**/secrets.dev.yaml
+**/values.dev.yaml
+**/build
+**/dist
+LICENSE
+README.md
```
Comment (gjbranham) [inline .dockerignore:28]: Did you add all these manually?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+APPLICATION ?= book-werm
+IMAGE ?= ${DOCKER_USERNAME}/${APPLICATION}
+
+
+build:
+	docker build -t ${IMAGE} .
+
+run:
+	docker run --name ${APPLICATION} -d -p ${PORT} ${IMAGE}
+	@echo "Running ${IMAGE} on ${PORT}, or open through Docker Desktop"
```
Comment (gjbranham) [inline Makefile:12]: Does this work without `PORT` and `DOCKER_USERNAME` filled out?

**Review by gjbranham — COMMENTED** [review]
Comments

File: .dockerignore (hunk 1 above)
Comment (hendraustin) [reply .dockerignore:28]: Ah, no, these were automatically added with `docker init`.

File: Makefile (hunk 2 above)
Comment (hendraustin) [reply Makefile:12]: Good point, will add logic to default that command when running without `PORT` or `DOCKER_USERNAME`.

Comment (hendraustin) [reply Makefile:11]: Ended up just setting default values that can be changed whenever by the developer for local testing.

Comment (gjbranham) [reply Makefile:11]: There are some other options you can do for default values. Maybe use an environment variable that defaults to an empty string or maybe the Makefile itself can set the environment variable to a default value. But this should work for now.

**Review by gjbranham — APPROVED** [review]

**Pull request merged**